### PR TITLE
Refine API

### LINKED ISSUE #15: Refine API
The current API feels clunky and uneasy to use. Something like the following could be simpler:

```
let command = Command(launchPath: launchPath,arguments: [])

let executor = CommandExecutor()

executor.outputHandler = {output, exitCode in
    // ...
    if exitCode == .commandNotFound {
        popup("Failed to run test")
    }
}

executor.write("foo")

executor.execute(command)

executor.terminate()
```

### PR DESCRIPTION
**Pull request by q231950**
- Add Command struct
- Allow command executor to be used in a fashion described in #15 

Fixes #15

### CODE REVIEW

File: Sources/CommandsCore/Command.swift [hunk 1]
```diff
@@ -0,0 +1,9 @@
+public struct Command {
+    let launchPath: String
+    let arguments: [String]
+
+    init(launchPath: String, arguments: [String] = []) {
+        self.launchPath = launchPath
+        self.arguments = arguments
+    }
+}
```
Comment (BasThomas) [inline Sources/CommandsCore/Command.swift]: Nit, but let's add a newline here.

File: Sources/CommandsCore/CommandExecutor.swift [hunk 2]
```diff
@@ -8,22 +8,17 @@
 import Foundation
 
 public class CommandExecutor {
+
+    public
+    var outputHandler: ((String) -> ())?
```
Comment (BasThomas) [inline Sources/CommandsCore/CommandExecutor.swift:13]: 💯

File: Tests/CommandsCoreTests/CommandExecutorTests.swift [hunk 3]
```diff
@@ -10,19 +10,34 @@ import XCTest
...
+        let command = Command(launchPath: "/bin/echo", arguments: ["hello world"])
+        XCTAssertNoThrow(commandExecutor.execute(command))
     }
 
-    func testCommandExecutorLaunchpath() {
-        let commandExecutor = CommandExecutor(launchPath: "/bin/echo", arguments: [])
-        XCTAssertEqual(commandExecutor.launchPath, "/bin/echo")
+    func testCommandExecutorExecutesCommand() {
+        let command = Command(launchPath: "/bin/echo")
+        XCTAssertNoThrow(commandExecutor.execute(command))
```
Comment (BasThomas) [inline Tests/CommandsCoreTests/CommandExecutorTests.swift]: When would this `throw`? The [function signature](https://github.com/q231950/commands/pull/17/files#diff-5dde104951dd9021903c742058d07ed8R19) is not marked with `throws`.

File: Sources/CommandsCore/Commands.swift [hunk 4]
```diff
@@ -13,15 +13,16 @@ public final class Commands {
         }
 
         let launchPath = arguments[1]
-        let commandExecutor = CommandExecutor(launchPath: launchPath,
-                                              arguments: forwardArguments(),
-                                              outputStream: StandardOutOutputStream())
+        let command = Command(launchPath: launchPath, arguments: forwardArguments())
+        let commandExecutor = CommandExecutor()
+        commandExecutor.outputHandler = { (text: String) in
```
Comment (BasThomas) [inline Sources/CommandsCore/Commands.swift]: Let's omit the `: String` and just use `{ text in` to be consistent.

**Review by BasThomas — APPROVED** [review]
Solid. Looks good to me.

File: Tests/CommandsCoreTests/CommandExecutorTests.swift (hunk 3 above)
Comment (q231950) [reply Tests/CommandsCoreTests/CommandExecutorTests.swift]: This is a good question. I'll have a look.

**Pull request merged**